### PR TITLE
Workspace upgrade (dependencies) & gitignore sync

### PR DESCRIPTION
**Pull request by ened**
In preparation for #760, I wanted to extract the dependency upgrades & improvements needed.
This PR:
 - updates the lock files from running `pub upgrade` everywhere
 - removes gradle wrapper binaries, in line with how new flutter projects are created (`flutter create -t app ..`)
 - upgrades Android/detekt to reduce security warnings 
 - upgrades Android/protobuf to reduce gradle warnings
 - minimally upgrades formatting to comply with detekt/ktlint configs
 - removes `.ruby-version`, a `rbenv` related file, as it is not relevant to the project (it's a detail on developer machines)

In order to get `analyzer` bumped to the latest version (and remove warnings etc), this [PR](https://github.com/spkersten/dart_functional_data/pull/36) needs to get merged in functional_data first & this project would then bump the dependency as needed.

### CODE REVIEW

**Comment by ened** [comment]
@Taym95 appreciate a review if you have a minute :-)

File: .editorconfig [hunk 1]
```diff
@@ -1,3 +1,2 @@
 [*.{kt,kts}]
-max_line_length=210
```
Comment (Taym95) [inline .editorconfig:2]: not needed?

**Review by Taym95 — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (ened) [reply .editorconfig:2]: fixed as part of this PR (to make all the tools happy), see the few kt changes

**Comment by ened** [comment]
I omitted a change log as no consumer-relevant changes are included this time, just build time/formatting improvements at most.

**Comment by ened** [comment]
@Taym95 let me know if this needs further adjustments

**Comment by ened** [comment]
Thank you @Taym95 - soo much cleaner over in #760 :)

**Pull request merged**